### PR TITLE
target/riscv: Handle sbbusyerror in read_memory_bus_v1

### PR DESCRIPTION
**Pull request by timsifive**
The existing code didn't seem to work right at all. I have spike modifications that exercise these new cases. I'll merge those once this has merged.

Maybe this helps with #880.

Change-Id: I89bd336f34f1b208a76f25b6b41fe3877800765b

### CODE REVIEW

**Comment by JanMatCodasip** [comment]
I will be able to take a closer look Friday morning.

File: src/target/riscv/riscv-013.c [hunk 1]
```diff
@@ -3195,7 +3197,17 @@ static int read_memory_bus_v1(struct target *target, target_addr_t address,
 					}
 					keep_alive();
 					dmi_status_t status = dmi_scan(target, NULL, &sbvalue[next_read_j],
-												   DMI_OP_READ, sbdata[j], 0, false);
+								       DMI_OP_READ, sbdata[j], 0, false);
+					if (j == 0 && info->bus_master_read_delay) {
```
Comment (JanMatCodasip) [inline src/target/riscv/riscv-013.c:3203]: This part was harder for me to understand:
```suggestion
					/* By reading from sbdata0, we have just initiated another system bus read.
					 * Wait to let the read finish. */
					if (j == 0 && info->bus_master_read_delay) {
```

**Review by JanMatCodasip — DISMISSED** [review]
I have checked this change visually. The fixes look all right to me, thank you.

@timsifive Do I understand correctly that you aim to add some sort of "system bus stress mode" into Spike?

File: src/target/riscv/riscv-013.c (hunk 1 above)
Comment (timsifive) [reply src/target/riscv/riscv-013.c:3203]: Done. Thanks.

Yes, I have spike changes that exercise this new code. I'm working on cleaning them up and will probably submit them over there next week.

**Review by JanMatCodasip — APPROVED** [review]

**Pull request merged**